iis-common/src/main/java/eu/dnetlib/iis/common/report/PigCountersReportGenerator.java

```diff
@@ -0,0 +1,116 @@
+package eu.dnetlib.iis.common.report;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import com.google.common.collect.Lists;
+
+import eu.dnetlib.iis.common.counter.PigCounters;
+import eu.dnetlib.iis.common.counter.PigCountersParser;
+import eu.dnetlib.iis.common.java.PortBindings;
+import eu.dnetlib.iis.common.java.Process;
+import eu.dnetlib.iis.common.java.io.DataStore;
+import eu.dnetlib.iis.common.java.io.FileSystemPath;
+import eu.dnetlib.iis.common.java.porttype.AvroPortType;
+import eu.dnetlib.iis.common.java.porttype.PortType;
+import eu.dnetlib.iis.common.schemas.ReportParam;
+
+/**
+ * Java workflow node process for building report from pig counters.<br/>
+ * <br/>
+ * It writes report properties into avro datastore of {@link ReportParam}s
+ * with location specified in output port.<br/>
+ * Report property key must start with <code>report.</code> to
+ * be included in output datastore.<br/>
+ * <br/>
+ * Report property values can contain placeholders for easier evaluation of
+ * pig counters. Placeholders are resolved using {@link PigCounterValueResolver}.<br/>
+ * <br/>
+ * Process needs <code>pigCounters</code> property that contains json representation
+ * of pig counters for working.<br/>
+ * 
+ * @author madryk
+ */
+public class PigCountersReportGenerator implements Process {
+
+    private static final String REPORT_PORT_OUT_NAME = "report";
+    
+    private static final String REPORT_PROPERTY_PREFIX = "report.";
+    
+    private static final String PIG_COUNTERS_PROPERTY = "pigCounters";
+    
+    
+    private PigCountersParser pigCountersParser = new PigCountersParser();
+    
+    private PigCounterValueResolver pigCounterValueResolver = new PigCounterValueResolver();
+    
+    
+    //------------------------ LOGIC --------------------------
+    
+    @Override
+    public Map<String, PortType> getInputPorts() {
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public Map<String, PortType> getOutputPorts() {
+        return Collections.singletonMap(REPORT_PORT_OUT_NAME, new AvroPortType(ReportParam.SCHEMA$));
+    }
+    
+    @Override
+    public void run(PortBindings portBindings, Configuration conf, Map<String, String> parameters) throws Exception {
+        
+        String pigCountersJson = parameters.get(PIG_COUNTERS_PROPERTY);
+        
+        PigCounters pigCounters = pigCountersParser.parse(pigCountersJson);
+        
+        
+        Map<String, String> reportParams = collectReportParameters(parameters);
```

Comment:
I would name this map something like 'reportCounterNames' and give some explanation that it is the map of pig counter names to human readable report names. It was hard for me to understand what it is.